lib/api-sync.js

```diff
@@ -81,16 +81,38 @@ function processSyncAPI(datasetId, params, readDatasetClient, cb) {
       return callback();
     },
     addAcks: function(callback) {
-      debug('[%s] adding acks to queue. size = %d', datasetId, (params.acknowledgements && params.acknowledgements.length || 0));
-      addToQueue(params.acknowledgements, {datasetId: datasetId, cuid: cuid}, ackQueue, callback);
+      var acknowledgements = params.acknowledgements || [];
+      debug('[%s] found acks in request. size = %d', datasetId, acknowledgements.length);
+      if (syncConfig.syncReqAckLimit && syncConfig.syncReqAckLimit > 0 && acknowledgements.length > syncConfig.syncReqAckLimit) {
+        acknowledgements = acknowledgements.slice(0, syncConfig.syncReqAckLimit);
```

Comment:
If we use default value here this statement may look like
`["ack1","ack2"].slice(0,-1)` => `["ack1"]` is that expected?